tracer/src/Datadog.Trace/Activity/OtlpHelpers.cs

```diff
@@ -204,6 +203,95 @@ private static void AgentConvertSpan<TInner>(TInner activity, Span span)
             {
                 span.Type = activity5 is null ? SpanTypes.Custom : AgentSpanKind2Type(activity5.Kind, span);
             }
+
+            // extract any ActivityLinks
+            ExtractActivityLinks<TInner>(span, activity5);
+        }
+
+        private static void ExtractActivityLinks<TInner>(Span span, IActivity5? activity5)
+            where TInner : IActivity
+        {
+            if (activity5 is null)
+            {
+                return;
+            }
+
+            foreach (var link in (activity5.Links))
+            {
+                if (link.TryDuckCast<IActivityLink>(out var duckLink))
+                {
+                    if (duckLink.Context.TraceId.TraceId is null || duckLink.Context.SpanId.SpanId is null)
+                    {
+                        continue;
+                    }
```

Comment:
nit: purely for readability I would do combine these (and reverse the first condition). 100% not required or necessary for this PR :D 
```suggestion
                if (!link.TryDuckCast<IActivityLink>(out var duckLink)
                    || duckLink.Context.TraceId.TraceId is null 
                    || duckLink.Context.SpanId.SpanId is null)
                    {
                        continue;
                    }
```